### DISCUSSION TITLE
Multiple entry points - how to keep one as small as possible?

### DISCUSSION THREAD

**Discussion by wardellbagby** [webpack/webpack#15564] [Q&A]
I've got a single-page application that shows a splash screen while the app loads. You can see it [here.](https://lyricistant.app). Due to naming being hard, that splash screen is named `preload`, while my actual application is named `main`.

My webpack config looks a bit like this:

```js
{
  entry: {
    preload: './src/preload.js',
    main: './src/main.js',
  },
}
```

My HTML looks a bit like this:

```html
<html>
  <body>
    <div id="app"></div>
    <script src="dist/preload.js"></script>
    <script async src="dist/main.js"></script>
  </body>
</html>
```

I've recently come to find out that since I'm including two entries in the same HTML file, I should _actually_ be setting `optimization.runtimeChunk` to `single` to avoid both of these entry chunks unnecessarily including the runtime. However, when I do that, my `dist/preload.js` balloons from 6kb to 30kb. Not a _huge_ gain but I want to keep that `dist/preload.js` as **small** as possible, since I need it to load first before my app can display anything.

Honestly, 6kb is already a lot for my `dist/preload.js` as my `src/preload.js` has 2 import: 1 that is a very simple function that returns a basic object and depends on no other imports, and a 2kb (unminified) SVG that's loaded with raw loader. The function IS used across both my preload and main entry points, though. Still, I was expecting more like 3kb as the code is so minimal but 6 has been good enough.

It seems like I've been getting lucky that my preload has even come out to 6kb; when I look at it, I see that it contains some code that my preload entry point never imports or uses, and that I know the single function I import never uses. Namely, it creates some symbols for various React stuff and some mappings of enum names to values.

Anyway, what I'm wondering is this: is there something I can change about my Webpack config to both take advantage of `runtimeChunks: 'single'` while also telling Webpack directly (instead of just getting lucky) that it should try and keep `dist/preload.js` as small as it possible can? I've tried `optimization.splitChunks.maxSize`, `optimization.splitChunks.maxInitialSize`, and `optimization.concatenateModules` to no avail.

Note: Everything here is a simplified version of my actual application but I've tried to be as detailed as possible while leaving out the irrelevant complexities of my application. However, my application is open-source and the code for it is [available here.](https://github.com/wardellbagby/lyricistant)

**Comment by vankop** [comment]
you can use `dependOn` option https://webpack.js.org/configuration/entry-context/#dependencies

**Reply by steverep** [reply]
I've got a similar situation with 2 entrypoints on a single page:
1. An 18 KB script that should load and run as fast as possible because it starts a web socket to begin downloading various data to be displayed
2. Much larger application code defining all the web components, etc.

The runtime is not currently being split out so as not to slow down the WS script.  Is `dependOn` the best way to handle this?  Can you spell out what that should look like?  Is there a single entry setup that could work the same or better?
